### PR TITLE
jnp.mean: fix normalizer for large arrays

### PR DESCRIPTION
**Pull request by jakevdp**
For large sharded arrays, it's currently possible for the mask sum to overflow int32, leading to strange outputs. Explicitly accumulating the sum using the computation dtype avoids this kind of overflow.

No unit tests here, because testing this would require more memory than is available in our test runners.

### CODE REVIEW

**Review by levskaya — APPROVED** [review]
thank you!
